hack/lib/init.sh

```diff
@@ -50,8 +50,10 @@ os::log::stacktrace::install
 # them before every invocation.
 os::util::environment::update_path_var
 
-if [[ -z "${OS_TMP_ENV_SET-}" ]]; then
-	os::util::environment::setup_tmpdir_vars "$( basename "$0" ".sh" )"
+if [[ -z "${OS_TMP_ENV_SET-}" ]] && [[ "$0" =~ *.sh ]]; then
```

Comment:
Hmm I think in this case we are still not having the same logic -- would want:

```sh
if [[ -z "${OS_TMP_ENV_SET-}" ]]; then
    if [[ "$0" =~ *.sh ]]; then
       os::util::environment::setup_tmpdir_vars "$( basename "$0" ".sh" )"
    else
        os::util::environment::setup_tmpdir_vars "shell"
    fi
fi
```

Reply:
thanks @stevekuznetsov , i have been done.